### PR TITLE
Guard sync completion paths against `Promise`-returning callbacks

### PR DESCRIPTION
**Pull request by dahlia**
Sync runner APIs (`runParser()`, `runParserSync()`, `runWithSync()`) already reject Promise-returning callbacks on help and parse-error paths by throwing `RunParserError("Synchronous parser returned async result.")`. However, the completion path in *handleCompletion()* bypassed this guard because it returns early before reaching the main `dispatchByMode` block that contains the check. As a result, if a completion callback (`onShow` or `onError`) returned a `Promise`, the sync APIs would silently return a live `Promise` object instead of throwing.

For example, the following code would return `Promise { "async-show" }` instead of throwing:

```typescript
runParser(parser, "myapp", ["completion", "bash"], {
  completion: {
    command: true,
    onShow: () => Promise.resolve("async-show") as unknown as string,
  },
  stdout: () => {},
  stderr: () => {},
});
```

This change adds `instanceof Promise` guards to all four `dispatchByMode` sync branches inside *handleCompletion()* in *packages/core/src/facade.ts*: the missing-shell error path, the unsupported-shell error path, the script generation path, and the suggestions path. Each now throws the same `RunParserError` that the help/error paths use.

## Test plan

- [x] Added 5 regression tests in *packages/core/src/facade.test.ts* covering all affected paths (script generation, suggestions, missing shell, unsupported shell, completion option mode)
- [x] Verified tests fail before the fix and pass after
- [x] `mise test` passes across Deno, Node.js, and Bun

Closes https://github.com/dahlia/optique/issues/264

### CODE REVIEW

File: packages/core/src/facade.test.ts [hunk 1]
```diff
@@ -9112,6 +9112,92 @@ describe("branch coverage: facade.ts edge cases", () => {
...
+          completion: {
+            command: true,
+            onShow: () => Promise.resolve("async-show") as unknown as string,
+          },
+          stdout: () => {},
+          stderr: () => {},
+        }),
+      RunParserError,
+      "Synchronous parser returned async result.",
+    );
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.test.ts:9132]: Both points are valid — fixed in 3688322. Used object matchers (`{ name, message }`) so the thrown error message is actually validated, and replaced `as unknown as string` with `as never`.

File: packages/core/src/facade.test.ts [hunk 2]
```diff
@@ -9112,6 +9112,92 @@ describe("branch coverage: facade.ts edge cases", () => {
...
 
+  it("throws when sync completion onShow returns a Promise (script generation)", () => {
+    // Regression test for https://github.com/dahlia/optique/issues/264
+    const parser = object({});
+    assert.throws(
+      () =>
+        runParser(parser, "myapp", ["completion", "bash"], {
+          completion: {
+            command: true,
+            onShow: () => Promise.resolve("async-show") as unknown as string,
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.test.ts]: Fixed in 3688322 alongside the object matcher change.

**Comment by dahlia** [comment]
@coderabbitai full review

File: packages/core/src/facade.ts [hunk 3]
```diff
@@ -1699,7 +1699,13 @@ function handleCompletion<M extends Mode, THelp, TError>(
...
-      () => callOnError(1),
+      () => {
+        const result = callOnError(1);
+        if (result instanceof Promise) {
+          throw new RunParserError("Synchronous parser returned async result.");
+        }
+        return result;
+      },
       () => Promise.resolve(callOnError(1)),
     );
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/facade.ts:1711]: Fixed in 467a129. Replaced `() => Promise.resolve(callOnError(1))` and `() => Promise.resolve(callOnCompletion(0))` with `async () => callOnError(1)` and `async () => callOnCompletion(0)` so synchronous throws become rejected promises. Added `deno-lint-ignore require-await` with reason comments since the async keyword is intentional here.

**Comment by dahlia** [comment]
@coderabbitai full review
